src/sst/core/baseComponent.cc

```diff
@@ -535,188 +536,164 @@ uint8_t BaseComponent::getComponentInfoStatisticEnableLevel(const std::string& s
 }
 
-
-StatisticBase*
-BaseComponent::registerStatisticCore(SST::Params& params, const std::string& statName, const std::string& statSubId,
-                                     fieldType_t fieldType, CreateFxn&& create)
-{
-    SST::Params statParams = params.find_prefix_params(statName);
-    std::string                     fullStatName;
-    bool                            statGood = true;
-    bool                            nameFound = false;
-    StatisticBase::StatMode_t       statCollectionMode = StatisticBase::STAT_MODE_COUNT;
-    Output &                        out = getSimulation()->getSimulationOutput();
-    StatisticProcessingEngine      *engine = StatisticProcessingEngine::getInstance();
-    UnitAlgebra                     collectionRate;
-    std::string                     statRateParam;
-    std::string                     statTypeParam;
-    StatisticBase*                   statistic = nullptr;
-
-
-    // First check to see if this is an "inserted" statistic that has
-    // already been created.
-
-    // Don't need to look at my cache, that was already done.  Start
-    // with my parent.
-    ComponentInfo* curr_info = my_info;
-    while (curr_info->canInsertStatistics() ) {
-        // Check to see if this was already created, and
-        // if so return the cached copy
-        StatisticBase* prevStat = StatisticProcessingEngine::getInstance()->isStatisticRegisteredWithEngine(
-            curr_info->parent_info->getName(), curr_info->parent_info->getID(), statName, statSubId, fieldType);
-        if (nullptr != prevStat) {
-            // Dynamic cast the base stat to the expected type
-            return prevStat;
-        }
-        curr_info = curr_info->parent_info;
-    }
-
-
-    // Build a name to report errors against
-    fullStatName = StatisticBase::buildStatisticFullName(getName().c_str(), statName, statSubId);
-
-    // Make sure that the wireup has not been completed
-    if (true == getSimulation()->isWireUpFinished()) {
-        // We cannot register statistics AFTER the wireup (after all components have been created)
-        out.fatal(CALL_INFO, 1, "ERROR: Statistic %s - Cannot be registered after the Components have been wired up.  Statistics must be registered on Component creation.; exiting...\n", fullStatName.c_str());
+void
+BaseComponent::configureCollectionMode(Statistics::StatisticBase* statistic, const SST::Params& params, const std::string& name)
+{
+  StatisticBase::StatMode_t statCollectionMode = StatisticBase::STAT_MODE_COUNT;
+  Output& out = getSimulation()->getSimulationOutput();
+  std::string statRateParam = params.find<std::string>("rate", "0ns");
+  UnitAlgebra collectionRate(statRateParam);
+
+  //make sure we have a valid collection rate
+  // Check that the Collection Rate is a valid unit type that we can use
+  if (collectionRate.hasUnits("s") || collectionRate.hasUnits("hz")) {
```

Comment:
I don't think I want to check both hz and Hz - shouldn't unit algebra already be able to handle case-sensitivity already? It seems like we should fix UnitAlgebra rather than rely on every unit check to have the correct case.

Reply:
I can't easily make UnitAlgebra do a case insensitive search.  The core Units code doesn't know about any actual units, they are added with static initialization later on (in fact, element libraries can add their own units if they want to).  I can't just obliviously do case insensitive search because you can't tell the m and M SI prefixes apart or the b and B units apart.  However, Hz is actually a "compound unit", which actually gives you 1/s once it's parsed.  You could just use hasUnits("1/s") and it will catch both cases.

Reply:
For completeness, I should note that there are two special cases in the SI prefix.  You can do either k or K for kilo and ki or Ki for kibi, even though K and Ki are technically wrong.  You can also do Hz and hz, even though hz is technically wrong,  They are misused enough, that I special cased both of them.  In the case of Hz/hz, I just added both as a compound unit (this special case would have been hard to do if Hz was a base unit instead of a compound unit).

Reply:
On further thought, because Hz is a compound unit, the code is fine as written.  The "hz" will translate to "1/s" and will check fine no matter how it was originally spelled.  Carry on.